src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoCalendar.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class DateTimeFormatInfoCalendar
+    {
+        [Fact]
+        public void Calendar_InvariantInfo()
+        {
+            Assert.Equal(typeof(GregorianCalendar), DateTimeFormatInfo.InvariantInfo.Calendar.GetType());
+        }
+
+        [Fact]
+        public void Calendar_Set()
+        {
+            Calendar newCalendar = new GregorianCalendar(GregorianCalendarTypes.Localized);
+            var format = new DateTimeFormatInfo();
+            format.Calendar = newCalendar;
+            Assert.Equal(newCalendar, format.Calendar);
+
+            format.Calendar = newCalendar;
+            Assert.Equal(newCalendar, format.Calendar);
```

Comment:
I'm the code, there is a fast return that occurs if the user sets the calendar to the same calendar